src/murrine_style.c

```diff
@@ -1182,6 +1182,22 @@ murrine_style_draw_box (DRAW_ARGS)
 
 		murrine_set_widget_parameters (widget, style, state_type, &params);
 
+		if (murrine_style->trough_use_child_bg && widget->parent && MRN_IS_SCROLLED_WINDOW (widget->parent)) {
+			GdkColor *gcolor = NULL;
+			GtkWidget *child  = gtk_bin_get_child ((GtkBin*) widget->parent);
```

Comment:
>  Can you explain what's happening here? Wouldn't gtk_bin_get_child (widget->parent) just return widget?

Sure, its the widget inside the ScrolledWindow. So if the ScrolledWindow has a child, params.parentbg (which is the background color of the parent container at this point) is set to its background color.

> Also, instead of casting (GtkBin*)widget->parent you should use the casting macros from gtk and do GTK_BIN
> (widget->parent). So, gtk_bin_get_child (GTK_BIN (widget->parent))

I don't think that a cast is required at all, because we won't reach here if `MRN_IS_SCROLLED_WINDOW (widget->parent) == FALSE` and GtkScrolledWindow is always a GtkBin (even if not `gtk_bin_get_child` would return `NULL` for widgets other than GtkBin, which is checked in the line below).

I did something with C a long time ago, so to be on the safe side I reused some patterns around. There are many redundant checks and casts in the code (like in https://github.com/mono/xamarin-gtk-theme/pull/4/files#diff-b6c2b3d97ebf86fb8ee24327d91af947R1185: the `widget->parent` check is not necessary and many more). So it could require some general revision. I'll push a cleanup of this part.